検索画面のUIの実装

nukomaru0930 opened the pull request:
<!-- GitHub Copilot コードレビューへの指示: このプルリクエストをレビューしてコメントするときは日本語でお願いします。 -->

#104 検索画面のUIの実装

<!-- 関連するIssueやチケットのリンクを貼る。close #イシュー番号でPRが閉じたタイミングでイシューを閉じれる -->

feature#104

close #104

## やったこと

<!-- このPRで何をしたのか？ -->

検索画面で使用するコンポーネントの実装
検索画面の実装
12/19追記　検索結果の作成

## やらないこと

バックエンドがまだなので検索履歴のロジックはまだできていない。

## できるようになること（ユーザ目線）

投稿の検索と投稿の絞り込みが可能

## できなくなること（ユーザ目線）

- 何ができなくなるのか？（あれば。無いなら「無し」で OK）

## 動作確認

投稿履歴の動作とレスポンシブ対応によるモーダルの動作確認

## 影響範囲

<!-- 影響を及ぼす範囲や他の機能への影響 -->

無し

## テスト

<!-- テスト方法や結果 -->

ダミーデータを用いた履歴の確認とチェックボックスの動作確認

## 備考

<!-- レビュワーへの伝達事項や残しておきたい情報 -->

検索履歴にダミーデータあり

nukomaru0930 committed:
feat: 検索画面の実装とそれに合わせたコンポーネントの作成　#104

nukomaru0930 committed:
feat:検索結果画面の作成 #104

nukomaru0930 committed:
refactor:レビュー内容の修正とtest.tsxのデータ削除feature/#104

hmuto111 commented on .vscode/settings.json:
```diff
@@ -1,5 +1,12 @@
+  "workbench.tree.renderIndentGuides": "always",
+  "workbench.colorCustomizations": {
+    "tree.indentGuidesStroke": "#ff0000" // Default is #555555.
+  },
```
この設定は個人設定だからコミットには含めないでください
前の状態にしてください

hmuto111 commented on frontend/src/app/routes/app/searchPost.tsx:
```diff
@@ -0,0 +1,64 @@
+  { id: 10, name: "e", classid: 9 },
+];
+
+const SearchPost = () => {
```
この指摘については無視でいいです。

hmuto111 commented on frontend/src/components/ui/search/search.module.css:
```diff
@@ -0,0 +1,38 @@
+  position: absolute;
+  top: 50%;
+  left: 15px;
+  pointer-events: none;
```
確認お願いします

hmuto111 commented on frontend/src/components/ui/search/search.module.css:
```diff
@@ -0,0 +1,68 @@
+    pointer-events: none;
+    font-size: 1.2rem;
+    color: rgb(50, 85, 150);
+    transform: translateY(-50%);
```
@mediaではプロパティに変化がない場合は、書かなくていいので不必要なものは消してください。
例：position: absoluteはメディアクエリ外で宣言されているので消すなど

hmuto111 commented on frontend/src/components/ui/search/search.module.css:
```diff
@@ -0,0 +1,68 @@
+    font-size: 1.2rem;
+    transition: transform 0.3s;
+    box-sizing: border-box;
+  }
```
searchIconと同様にこちらも不必要なものは消してください。

hmuto111 commented on frontend/src/components/ui/search/search.tsx:
```diff
@@ -0,0 +1,24 @@
+        placeholder="検索"
+        className={styles.searchInput}
+      />
+      <FaSearch onClick={onSearch} className={styles.searchIcon} />
```
こちらも確認お願いします

hmuto111 commented on frontend/src/features/search/components/searchFilter.module.css:
```diff
@@ -0,0 +1,55 @@
+
+.checkIcon {
+  appearance: none;
+  -webkit-appearance: none;
```
これって何用のプロパティですか？

hmuto111 commented on frontend/src/features/search/components/searchFilter.tsx:
```diff
@@ -0,0 +1,230 @@
+          textAlign: "center",
+          userSelect: "none",
+          padding: "5px",
+          visibility: hasChildren ? "visible" : "hidden",
```
ここスタイル直書きできればクラス指定のほうがいい？

hmuto111 commented on frontend/src/features/search/components/searchFilter.tsx:
```diff
@@ -0,0 +1,231 @@
+
+  return (
+    <div>
+      <div>
+        <strong>aaa</strong>
+        <div>
+          {Array.from(selectedIds)
+            .sort((a, b) => Number(a) - Number(b))
+            .join(",") || "Global"}
+        </div>
+      </div>
```
ここのパンくずリストはたぶんまだ実装していないと思うんでロジック実装（データ取得とか）時に対応するといいと思います

hmuto111 commented on frontend/src/features/search/components/searchHistory.tsx:
```diff
@@ -0,0 +1,53 @@
+  "むとうはるき かわいい",
+  "むとうはるき いけめん",
+  "むとうはるき あたまいい",
+  "むとうはるき 何者？",
```
個人名はプライバシーにかかわるので消してください😡

hmuto111 reviewed (changes requested):
確認お願いします

nukomaru0930 committed:
refactor:jsonファイル前の形に修正。
searchHistory.tsxダミーデータ修正。
searchFilter.tsxスタイルをcssに移動しcss側は不要なプロパティ削除。
search.tsxボタン押下による検索イベントの追加。
cssは不要なプロパティ削除。

nukomaru0930 committed:
refactor:HTMLのデフォルトデザインを削除する処理追加しました feature/#104

hmuto111 commented on .vscode/settings.json:
```diff
@@ -1,5 +1,4 @@
 {
-  "extensions.ignoreRecommendations": false,
```
これもともとあったやつなんで消さなくていいです

hmuto111 reviewed (changes requested):
settings.jsonだけ確認してください

nukomaru0930 committed:
refactor:setting.jsonの中身消さなくてよかったデータ戻しました。feature/#104

hmuto111 reviewed (approved):
OKです
